feat(mcp): add Python agent example, integration tests, and CI

datradito opened the pull request:
Adds a minimal working example and test coverage for the MCP server
introduced in #1502.

## What's included

**`examples/mcp_agent.py`** — A LangGraph-based autonomous agent that
connects to a live Echidna campaign over MCP. Each iteration it:
1. Calls `status` to read coverage and corpus size
2. When coverage stagnates, asks Claude to generate targeted transactions
   and injects them via `inject_fuzz_transactions`
3. Periodically calls `clear_fuzz_priorities` to avoid getting stuck

**`examples/README.md`** — Setup, usage, and a reference table of all
available MCP tools.

**`tests/mcp/test_mcp.py`** — pytest integration tests that spin up a
real Echidna process and exercise the four core MCP operations:
inject transactions, check coverage, reset priorities, check status.

**`tests/mcp/contracts/`** — `EchidnaMCPTest.sol` and `SimpleToken.sol`,
the contracts used by the test suite.

**`.github/workflows/mcp-tests.yml`** — CI job that builds Echidna via
Nix and runs the integration tests on every push and PR.

## How to run locally

```bash
# Install test deps
pip install pytest httpx

# Run integration tests (requires Echidna binary in PATH)
pytest tests/mcp/test_mcp.py -v
```

## How to run the agent

```bash
# Terminal 1 — start Echidna with MCP enabled
echidna MyContract.sol --server 8080 --format text

# Terminal 2 — run the agent
export ANTHROPIC_API_KEY=your_key_here
python examples/mcp_agent.py
```

> `--format text` is required: it disables the interactive TUI so the
> MCP server thread can accept connections.

## Notes

- Depends on #1502 (the MCP server itself)
- The agent requires an Anthropic API key; the test suite does not

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/crytic/echidna?pullRequest=1509) <br/>All committers have signed the CLA.

datradito commented on examples/langgraph_agent.py:
```diff
@@ -0,0 +1,285 @@
+class EchidnaLangGraphAgent:
+    """Advanced LangGraph agent for Echidna MCP integration."""
+    
+    def __init__(self, mcp_url: str = "http://localhost:8080/mcp", model: str = "claude-3-5-sonnet-20241022"):
```
This is only for example purpose.

gustavo-grieco reviewed (changes requested):
I like the direction of this PR, here is a couple of comments:
* Why exactly we need to log all the commands to a file in the MCP?
* Start addressing all the copilot comments, discarding with a comment the ones that makes no sense.
* You added some basic testing code for the MCP using Python, that's a great, but we need to make sure it is used by our CI. Include a new CI test for that (which should run in parallel). If you have questions about that, @elopez is our Github Action expert.
* The added documentation is very useful. We usually have the documentation for tools like echidna in [building-secure-contracts](https://github.com/crytic/building-secure-contracts), but we usually update it when it is close to release. However, don't remove the documentation, we can keep it until we are more confident on how to use this agentic capabilities, and move them into building secure contracts when we are close to release.

datradito committed:
feat(mcp): add Python example, integration test, and CI

- examples/mcp_agent.py: LangGraph agent that observes coverage,
  injects targeted transactions, and resets priorities when stagnating
- examples/README.md: brief usage docs (start command, tool table,
  transaction format)
- tests/mcp/test_mcp.py: four integration tests covering the core
  workflow: status, inject_fuzz_transactions, show_coverage,
  clear_fuzz_priorities
- tests/mcp/contracts/: EchidnaMCPTest + SimpleToken for test campaigns
- .github/workflows/mcp-tests.yml: CI job that builds echidna via Nix
  and runs pytest tests/mcp/test_mcp.py on every push/PR

datradito committed:
docs(mcp): expand test requirements with full langchain/langgraph deps

Adds pinned versions for langchain, langgraph, pytest-asyncio, jsonschema,
and python-dotenv so developers can install a complete local environment
for the MCP agent example. CI is unaffected (it installs pytest and httpx
directly).

gustavo-grieco commented:
This PR was squashed/vendored in the `dev-agents` branch directly: https://github.com/crytic/echidna/pull/1502/changes/1975190665b5e2d1b6672c60b1f5c00a056e847b
